pkg/controller/populators/upload-populator.go

```diff
@@ -159,13 +159,15 @@ func (r *UploadPopulatorReconciler) reconcileTargetPVC(pvc, pvcPrime *corev1.Per
 		// We'll get called later once it succeeds
 		r.recorder.Eventf(pvc, corev1.EventTypeWarning, errUploadFailed, fmt.Sprintf(messageUploadFailed, pvc.Name))
 	case string(corev1.PodSucceeded):
-		// Once the upload is succeeded, we rebind the PV from PVC' to target PVC
-		if err := cc.Rebind(context.TODO(), r.client, pvcPrime, pvcCopy); err != nil {
-			return reconcile.Result{}, err
+		if cc.IsPVCComplete(pvcPrime) {
```

Comment:
Same here, I think we should check here `&& cc.IsUnbound(pvc)`

Reply:
ACK